tests/lib/server.py

```diff
@@ -42,9 +43,12 @@ class MockServer(BaseWSGIServer):
     def blocked_signals():
         """Block all signals for e.g. starting a worker thread.
         """
-        old_mask = signal.pthread_sigmask(
-            signal.SIG_SETMASK, range(1, signal.NSIG)
-        )
+        if CURRENT_PY_VERSION_INFO < (3, 8, 0):
+            mask = range(1, signal.NSIG)
+        else:
+            mask = signal.valid_signals()
```

Comment:
Would it make sense to do

```python
try:
    mask = signal.valid_signals()
except AttributeError:
    mask = range(1, signal.NSIG)
```

or even

```python
mask = getattr(signal, valid_signals, functools.partial(range, 1, signal.NSIG))()
```

Reply:
the first one looks good to me, I can do that (although it needs an explanatory comment, mine should does not).

Are the tests flaky? It seems strange only windows/3.6/x86_64 fails...

Reply:
I think both would benefit from some kind of explainatory comment. The if-version check does not convey *why* `signal.valid_signals()` is not called on 3.7 and earlier (it’s not available, instead of being buggy).

The failing test on Windows is caused by a failure to remove temporary directories, which is unfortunately not uncommon.